### ISSUE TITLE
[JOSS REVIEW] Describe output of simulator_fn

### ISSUE THREAD

**Issue by aneeshnaik** [dirmeier/sbijax#91]
This issue is part of the JOSS review for sbijax. Review thread: https://github.com/openjournals/joss-reviews/issues/11195.

The new "workflow" sections describe the required inputs (theta and seed) of the simulator_fn. Would be good to also be specific about the required output format.

When setting up my [experimental test](https://gist.github.com/aneeshnaik/09ed93e428d0d0e9bd860ac4b23143b2) of `sbijax`, I was initially encountering an error, and it took me a while to understand that it was because my simulator function was returning arrays of shape (N_batch,) rather  than (N_batch, 1) (my example was 1D inference).
